src/DocumentFormat.OpenXml.Framework/Equality/OpenXmlElementEqualityComparer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace DocumentFormat.OpenXml.Equality
+{
+    /// <summary>
+    /// Equality comparer for determining value equality for <see cref="OpenXmlElement"/>.
+    /// </summary>
+    public sealed class OpenXmlElementEqualityComparer : IEqualityComparer<OpenXmlElement>
```

Comment:
Can this be a static class for now with internal class that implements the equality comparer?